src/Api.Rest/Common/Client/RestClientBase.cs

```diff
@@ -440,9 +512,24 @@ private static async Task<T> BinaryResponseToObjectAsync<T>( HttpResponseMessage
 						return default;
 					}
 
+					if( _AuthenticationHandler != null )

```

Comment:
Would it make sense to only do this when the response code is a 401?

Reply:
I see that this is done in the NonInteractiveAuthenticationHandler, but the user could use their own AuthHandler where this is not the case.

Reply:
It seems to me that it should be the job of an AuthenticationHandler to determine whether an error is actually authentication related or not. Since it also does not introduce a significant amount of complexity in the AuthenticationHandler, I opted for this approach.